[2.0] bmake: fix test issue

anphel31 opened the pull request:
<!--
COMMENT BLOCKS WILL NOT BE INCLUDED IN THE PR.
Feel free to delete sections of the template which do not apply to your PR, or add additional details
-->

###### Merge Checklist  <!-- REQUIRED -->
<!-- You can set them now ([x]) or set them later using the Github UI -->
**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary <!-- REQUIRED -->
<!-- Quick explanation of the changes. -->
Improve reliability of `bmake` package build by skipping a test that is known to be flaky (fails every week or two in daily builds).  
The unit tests currently run during the %install section - I've changed this, so the tests run in a %check section.

###### Change Log  <!-- REQUIRED -->
<!-- Detail the changes made here. -->
<!-- Please list any packages which will be affected by this change, if applicable. -->
<!-- Please list any CVES fixed by this change, if applicable. -->
- Change bmake to skip failing test

###### Does this affect the toolchain?  <!-- REQUIRED -->
<!-- Any packages which are included in the toolchain should be carefully considered. Make sure the toolchain builds with these changes if so. -->
<!-- Update: manifests/package/toolchain_*.txt, pkggen_core_*.txt, update_manifests.sh -->
<!-- To validate: make clean; make workplan REBUILD_TOOLCHAIN=y DISABLE_UPSTREAM_REPOS=y CONFIG_FILE="" ... -->
**NO**

###### Associated issues  <!-- optional -->
<!-- Link to Github issues if possible. -->
<!-- you can use "fixes #xxxx" to auto close an associated issue once the PR is merged -->
- https://microsoft.visualstudio.com/OS/_workitems/edit/57304584

###### Test Methodology
<!-- How was this test validated? i.e. local build, pipeline build etc. -->
- Buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=810848&view=results

sameluch reviewed (approved):
Approved, but is there a way to only run the tests as a part of a %check section instead of on install?

PawelWMS commented on SPECS/bmake/bmake.spec:
```diff
@@ -48,6 +48,8 @@ sh ./make-bootstrap.sh
 
 %install
 export STRIP=/bin/true # Make sure binary is not stripped
+# skip 'job-output-null' which randomly fails in build pipelines
+export BROKEN_TESTS=job-output-null
```
I think I've seen some logic inside the build files for `bmake`, where this test is already supposed to be skipped when the build runs in Bash (this should be the case here). Have you tried checking, why that doesn't trigger during our builds? Fixing that may be more future-proof.

Having said that, since 2.0 is getting deprecated, I think this is also good enough. I doubt we'll be updating this package.

anphel31 replied:
This test is automatically skipped for the `SCO_SV` OS but that's not triggered for AZL.

anphel31 commented:
> Approved, but is there a way to only run the tests as a part of a %check section instead of on install?

Good idea. Yes, I've added a patch to skip calling the unit tests during install, and I've moved them to a new %check section.
